src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/JenkinsAwareGerritHandler.java

```diff
@@ -28,29 +28,44 @@
 
 import com.sonymobile.tools.gerrit.gerritevents.GerritHandler;
 import com.sonymobile.tools.gerrit.gerritevents.dto.GerritEvent;
+import com.sonymobile.tools.gerrit.gerritevents.workers.Coordinator;
+import com.sonymobile.tools.gerrit.gerritevents.workers.EventThread;
 import com.sonyericsson.hudson.plugins.gerrit.trigger.events.lifecycle.GerritEventLifecycle;
 
 /**
- * Specialization of GerritHandler that supports gerrit event's lifecycle.
- * @author Hugo Arès &lt;hugo.ares@ericsson.com&gt;
+ * Specialization of GerritHandler that supports gerrit event's
+ * lifecycle and takes care of custom EventThread creation.
  *
+ * @author Hugo Arès &lt;hugo.ares@ericsson.com&gt;
  */
-public class GerritHandlerLifecycle extends GerritHandler {
+public class JenkinsAwareGerritHandler extends GerritHandler implements Coordinator {
```

Comment:
Renaming public classes like this will break backwards compatibility quite severely.

Although a rename is probably in order here the old class should probably be kept but deprecated.